### ISSUE TITLE
Out-DbaDataTable issues

### ISSUE THREAD

**Issue by potatoqualitee** [sqlcollaborative/dbatools#1485]
## Is this:
_Just place "x" to indicate selection (e.g. "[x]")_

- [ ] *feature* 
- [x] *bug*:

## Details
 If it's possible, it would be beyond helpful to get this fixed by Wednesday - I tested this and both fail on the last commit by @FriedrichWeinmann  and the previous commit by @JohnRoos 

First, just doing an `ls | Out-DbaDataTable` doesn't work:

````
Exception setting "Item": "Exception calling "set_Item" with "2" argument(s): "Column 'VersionInfo' does not belong to table ."
At line:246 char:37
+ ...                       $datarow.Item($property.Name) = $property.value
+                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], SetValueInvocationException
    + FullyQualifiedErrorId : CatchFromBaseAdapterParameterizedPropertySetValue

Exception setting "Item": "Exception calling "set_Item" with "2" argument(s): "Column 'Length' does not belong to table .""
````

Next, getting errors when trying to turn a datatable into a datatable

````
PS C:\github\dbatools> Get-DbaDatabase -SqlInstance sql2016 | Out-DbaDataTable
The following exception occurred while trying to enumerate the collection: "unknown property IsSystemObject".
At line:235 char:17
+                         if ($property.value.length -gt 0) {
+                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], ExtendedTypeSystemException
    + FullyQualifiedErrorId : ExceptionInGetEnumerator

The following exception occurred while trying to enumerate the collection: "unknown property IsSystemObject".
At line:235 char:17
+                         if ($property.value.length -gt 0) {
+                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], ExtendedTypeSystemException
    + FullyQualifiedErrorId : ExceptionInGetEnumerator

The following exception occurred while trying to enumerate the collection: "unknown property IsSystemObject".
````

**Comment by JohnRoos** [comment]
The first problem you are describing is because Get-ChildItem is returning two different types of objects, one for directories and one for files. Out-DbaDataTable requires all objects to be of the same type, otherwise it will show that kind of error. You can use for example "Get-ChildItem -File | Out-DbaDataTable" to store a list of files in a datatable instead.

The second issue should be fixed in pull request #1488 (which also include a small datetime overflow fix for Write-DbaDataTable)

**Comment by potatoqualitee** [comment]
ouuu thank you 😻 fantastic, man!
